### PR TITLE
Update Cart handling so it's not just REST context only

### PR DESCRIPTION
**Pull request by sc0ttkclark**
https://central.tri.be/issues/136611

### CODE REVIEW

**Comment by sc0ttkclark** [comment]
Related https://github.com/moderntribe/event-tickets-plus/pull/894

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "dev": "cross-env NODE_ENV=development webpack -d --watch",
     "gettext-strings": "cross-env BABEL_ENV=gettext webpack",
     "preanalyze": "cross-env NODE_ENV=production webpack -p --profile --json > stats.json",
-    "rebuild": "rm -rf node_modules && rm -rf vendor && composer update && npm install && npm run build",
```
[the remark this answers is not in the document]
Comment (Camwyn) [reply package.json]: I'm ok with removing this though.

[the remark this answers is not in the document]
Comment (sc0ttkclark) [reply package.json]: I figured that until we get it solved across the board, we should remove it so they all work the same (and we should get it added to common at some point)

File: src/Tribe/Commerce/Cart.php [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+			$provider_key             = $provider_object->orm_provider;
+			$provider_attendee_object = $provider_object->attendee_object;
+
+			// Skip provider if we only want specific ones.
+			if (
+				[] !== $providers
+				&& ! in_array( $provider_key, $providers, true )
+				&& ! in_array( $provider_attendee_object, $providers, true )
+				&& ! in_array( $provider_data['class'], $providers, true )
+			) {
```
[the remark this answers is not in the document]
Comment (sc0ttkclark) [reply src/Tribe/Commerce/Cart.php]: I prefer to look for `woo`, `tribe-commerce`, or `edd` but I noticed the values get interchanged a lot right now and we can remove support for the others as soon as we stop using the full class names everywhere.

[the remark this answers is not in the document]
Comment (sc0ttkclark) [reply src/Tribe/Commerce/Cart.php]: Multi provider cart functionality was just part of what I added initially, it's not meant to be used at this time.

File: src/Tribe/Commerce/Cart.php [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+				 * @since TBD
+				 *
+				 * @param string $checkout_url Checkout URL.
+				 * @param array  $data         Commerce response data to be sent.
+				 * @param int    $post_id      Post ID for the cart.
+				 */
+				$data['checkout_url'] = apply_filters( 'tribe_tickets_commerce_cart_get_checkout_url_' . $cart_provider, '', $data, $post_id );
+
+				// Stop after first provider URLs are set.
+				if ( '' !== $data['cart_url'] || '' !== $data['checkout_url'] ) {
```
[the remark this answers is not in the document]
Comment (sc0ttkclark) [reply src/Tribe/Commerce/Cart.php]: Right, if/when we support multiple providers in the future we can sort this part out. This is just the result of it being able to support multiple carts from PHP.

**Pull request merged**